src/io/iohk/decco/Codec.scala

```diff
@@ -13,6 +13,7 @@ trait Codec[T] { self =>
   final def encode[B](t: T)(implicit bi: BufferInstantiator[B]): B = {
     val bb = bi.instantiateByteBuffer(size(t))
     encodeImpl(t, 0, bb)
+    (bb: java.nio.Buffer).position(0)
```

Comment:
What if the byte buffer was in a position different to 0?

Reply:
We are creating this byte buffer, we are just putting it's internal pointer at the begining. All the problems scalanet was having were related with this